### PR TITLE
Mbasiuk/tdi 38933 sf integration tests

### PR DESCRIPTION
**Pull request by mbasiuk-talend**
**What is the current behavior?** (You can also link to an open issue here)
ITs for Salesforce Reader were unstable due to shared data.

**What is the new behavior?**
Create different data for every execution. Fix unstable test for session renewal.

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [x] Improve Salesforce Reader integration tests.

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/686/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/686/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/687/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/687/)

File: components/components-salesforce/components-salesforce-integration/src/test/java/org/talend/components/salesforce/runtime/SalesforceInputReaderTestIT.java [hunk 1]
```diff
@@ -219,63 +239,46 @@ public Schema getMakeRowSchema(boolean isDynamic) {
...
+        IndexedRecord fkRecord = rowsWithForeignKey.get(0);
+        IndexedRecord commonRecord = outputRows.get(0);
+        assertNotNull(fkRecord);
+        assertNotNull(commonRecord);
+        Schema schemaFK = fkRecord.getSchema();
+        Schema schemaCommon = commonRecord.getSchema();
+
+        assertNotNull(schemaFK);
+        assertNotNull(schemaCommon);
+        assertEquals(commonRecord.get(schemaCommon.getField("OwnerId").pos()), fkRecord.get(schemaFK.getField("Owner_Id").pos()));
```
Comment (dmytro-sylaiev) [inline components/components-salesforce/components-salesforce-integration/src/test/java/org/talend/components/salesforce/runtime/SalesforceInputReaderTestIT.java]: Please next time follow this pattern in big test-methods to increase readable of test code: 
http://wiki.c2.com/?ArrangeActAssert
Thank you

Comment (mbasiuk-talend) [inline components/components-salesforce/components-salesforce-integration/src/test/java/org/talend/components/salesforce/runtime/SalesforceInputReaderTestIT.java]: Thank you!

**Pull request merged**